Framework/Core/src/DataAllocator.cxx

```diff
@@ -181,6 +184,9 @@ void DataAllocator::adopt(const Output& spec, TreeToTable* t2t)
   std::shared_ptr<TreeToTable> p(t2t);
   auto finalizer = [payload = p](std::shared_ptr<FairMQResizableBuffer> b) -> void {
     auto table = payload->finalize();
+    if (O2_BUILTIN_UNLIKELY(table->num_rows() == 0)) {
+      LOG(WARN) << "Adopting empty table: " << table->ToString();
```

Comment:
I do not like this. Either an empty table is fine, or not. We should not "WARN" in that case, to me it means something which needs to be fixed but an empty table is not necessarily such.

Reply:
True, for TreeToTable, unlike TableBuilder, it makes more sense to either ignore it or throw an error. Should I just remove this or change to a throw?

Reply:
As discussed privately:

* we should fix the schema to be always there, also in case of empty tables, to avoid crashes.
* We can demote this to debug once it happens.

Merging.